server/core/dependency_injector.py

```diff
@@ -0,0 +1,137 @@
+import inspect
+from collections import ChainMap, defaultdict
+from typing import Dict, List
+
+DependencyGraph = Dict[str, List[str]]
+
+
+class DependencyInjector(object):
```

Comment:
imho, this code should be in separate ticket.

Reply:
Yea I guess that's reasonable. I don't think the dependency-injector package really does what I want. As far as I can tell with that package you still have to pass arguments in manually after you've defined the classes. Their code is all written in cython though and it's really hard to read.